### ISSUE TITLE
Failure on query disks on Dell PERC(megaraid)

### ISSUE THREAD

**Issue by cathay4t** [libstorage/libstoragemgmt#143]
Got a dell PERC system has no enclosure ID with failure:
```
[root@dell-per310-01 SPECS]# lsmcli ld -u megaraid://
PLUGIN_BUG(2): cmd: '/opt/MegaRAID/perccli/perccli64 /c0/eall/sall show all J', errno: 255, stdout: '{
"Controllers":[
{
        "Command Status" : {
                "Controller" : 0,
                "Status" : "Failure",
                "Description" : "No drive found!"
        }
}
]
}', stderr: ''
```

All disks could be queried by command `/opt/MegaRAID/perccli/perccli64 /c0/sall show all J`

```
[root@dell-per310-01 SPECS]# /opt/MegaRAID/perccli/perccli64 /c0/sall show all J
{
"Controllers":[
{
        "Command Status" : {
                "Controller" : 0,
                "Status" : "Success",
                "Description" : "Show Drive Information Succeeded."
        },
        "Response Data" : {
                "Drive /c0/s0" : [
                        {
                                "EID:Slt" : " :0",
                                "DID" : 0,
                                "State" : "Onln",
                                "DG" : 0,
                                "Size" : "148.5 GB",
                                "Intf" : "SATA",
                                "Med" : "HDD",
                                "SED" : "N",
                                "PI" : "N",
                                "SeSz" : "512B",
                                "Model" : "WDC WD1602ABKS-18N8A0",
                                "Sp" : "U"
                        }
                ],
                "Drive /c0/s0 - Detailed Information" : {
                        "Drive /c0/s0 State" : {
                                "Shield Counter" : 0,
                                "Media Error Count" : 0,
                                "Other Error Count" : 0,
                                "Drive Temperature" : "N/A",
                                "Predictive Failure Count" : 0,
                                "S.M.A.R.T alert flagged by drive" : "No"
                        },
                        "Drive /c0/s0 Device attributes" : {
                                "SN" : "WD-WCAT26034823",
                                "Manufacturer Id" : "ATA     ",
                                "Model Number" : "WDC WD1602ABKS-1",
                                "NAND Vendor" : "",
                                "WWN" : "50014ee1026ddfa1",
                                "Firmware Revision" : "3B04",
                                "Raw size" : "149.011 GB [0x12a05f20 Sectors]",
                                "Coerced size" : "148.5 GB [0x12900000 Sectors]",
                                "Non Coerced size" : "148.511 GB [0x12905f20 Sectors]",
                                "Device Speed" : "3.0Gb/s",
                                "Link Speed" : "3.0Gb/s",
                                "Sector Size" : "512B"
                        },
                        "Drive /c0/s0 Policies/Settings" : {
                                "Drive position" : "DriveGroup:0, Span:0, Row:0",
                                "Enclosure position" : 0,
                                "Connected Port Number" : "1(path0) ",
                                "Sequence Number" : 2,
                                "Commissioned Spare" : "No",
                                "Emergency Spare" : "No",
                                "Last Predictive Failure Event Sequence Number" : 0,
                                "Successful diagnostics completion on" : "N/A",
                                "SED Capable" : "No",
                                "SED Enabled" : "No",
                                "Secured" : "No",
                                "Locked" : "No",
                                "Needs EKM Attention" : "No",
                                "PI Eligible" : "No",
                                "Certified" : "Yes",
                                "Wide Port Capable" : "No",
                                "Port Information" : [
                                        {
                                                "Port" : 0,
                                                "Status" : "Active",
                                                "Linkspeed" : "3.0Gb/s",
                                                "SAS address" : "0x4433221107000000"
                                        }
                                ]
                        },
                        "Inquiry Data" : "7a 42 ff 3f 37 c8 10 00 00 00 00 00 3f 00 00 00 00 00 00 00 20 20 20 20 57 20 2d 44 43 57 54 41 36 32 33 30 38 34 33 32 00 00 00 80 32 00 32 30 30 2e 42 33 34 30 44 57 20 43 44 57 36 31 32 30 42 41 53 4b 31 2d 4e 38 41 38 20 30 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 10 80 00 00 00 2f 01 40 00 00 00 00 07 00 ff 3f 10 00 3f 00 10 fc fb 00 00 01 ff ff ff 0f 00 00 07 00 "
                },
                "Drive /c0/s1" : [
                        {
                                "EID:Slt" : " :1",
                                "DID" : 1,
                                "State" : "Onln",
                                "DG" : 0,
                                "Size" : "148.5 GB",
                                "Intf" : "SATA",
                                "Med" : "HDD",
                                "SED" : "N",
                                "PI" : "N",
                                "SeSz" : "512B",
                                "Model" : "WDC WD1602ABKS-18N8A0",
                                "Sp" : "U"
                        }
                ],
                "Drive /c0/s1 - Detailed Information" : {
                        "Drive /c0/s1 State" : {
                                "Shield Counter" : 0,
                                "Media Error Count" : 0,
                                "Other Error Count" : 0,
                                "Drive Temperature" : "N/A",
                                "Predictive Failure Count" : 0,
                                "S.M.A.R.T alert flagged by drive" : "No"
                        },
                        "Drive /c0/s1 Device attributes" : {
                                "SN" : "WD-WCAT26036741",
                                "Manufacturer Id" : "ATA     ",
                                "Model Number" : "WDC WD1602ABKS-1",
                                "NAND Vendor" : "",
                                "WWN" : "50014ee1ad11d86d",
                                "Firmware Revision" : "3B04",
                                "Raw size" : "149.011 GB [0x12a05f20 Sectors]",
                                "Coerced size" : "148.5 GB [0x12900000 Sectors]",
                                "Non Coerced size" : "148.511 GB [0x12905f20 Sectors]",
                                "Device Speed" : "3.0Gb/s",
                                "Link Speed" : "3.0Gb/s",
                                "Sector Size" : "512B"
                        },
                        "Drive /c0/s1 Policies/Settings" : {
                                "Drive position" : "DriveGroup:0, Span:0, Row:1",
                                "Enclosure position" : 0,
                                "Connected Port Number" : "0(path0) ",
                                "Sequence Number" : 2,
                                "Commissioned Spare" : "No",
                                "Emergency Spare" : "No",
                                "Last Predictive Failure Event Sequence Number" : 0,
                                "Successful diagnostics completion on" : "N/A",
                                "SED Capable" : "No",
                                "SED Enabled" : "No",
                                "Secured" : "No",
                                "Locked" : "No",
                                "Needs EKM Attention" : "No",
                                "PI Eligible" : "No",
                                "Certified" : "Yes",
                                "Wide Port Capable" : "No",
                                "Port Information" : [
                                        {
                                                "Port" : 0,
                                                "Status" : "Active",
                                                "Linkspeed" : "3.0Gb/s",
                                                "SAS address" : "0x4433221106000000"
                                        }
                                ]
                        },
                        "Inquiry Data" : "7a 42 ff 3f 37 c8 10 00 00 00 00 00 3f 00 00 00 00 00 00 00 20 20 20 20 57 20 2d 44 43 57 54 41 36 32 33 30 37 36 31 34 00 00 00 80 32 00 32 30 30 2e 42 33 34 30 44 57 20 43 44 57 36 31 32 30 42 41 53 4b 31 2d 4e 38 41 38 20 30 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 10 80 00 00 00 2f 01 40 00 00 00 00 07 00 ff 3f 10 00 3f 00 10 fc fb 00 00 01 ff ff ff 0f 00 00 07 00 "
                }
        }
}
]
}
```

**Comment by cathay4t** [comment]
Will fix in 1.4

**Comment by tasleson** [comment]
@cathay4t How difficult would this be to squeeze into 1.3 release?

**Comment by cathay4t** [comment]
@tasleson I will try to fix it today. AFAIK, only `pool_member_info()` and `disks()` impacted for this.

**Comment by cathay4t** [comment]
Closing. Fixed in #148.
